File: MoravaEngine/Projects/HangmanScript/src/CGL/GameConsole.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Runtime.CompilerServices;


namespace CGL
{
    public class GameConsole
    {
        [MethodImplAttribute(MethodImplOptions.InternalCall)]
        public extern static void clearBuffer();

        [MethodImplAttribute(MethodImplOptions.InternalCall)]
        public extern static void present();

        [MethodImplAttribute(MethodImplOptions.InternalCall)]
        public extern static void putChar(uint x, uint y, char c);

        [MethodImplAttribute(MethodImplOptions.InternalCall)]
        public extern static void putString(uint x, uint y, string str, bool wrap = true);

    }
}

File: MoravaEngine/src/Mono/ConsoleGameLib/GameConsole.cpp
#include "GameConsole.h"
#include "GConsole.h"


namespace CGL
{
	void GameConsole::clearBuffer()
	{
		// Get and check console pointer
		auto consolePointer = CGL_internal::GConsole::get();
		if (consolePointer)
		{
			// Redirect call
			consolePointer->clearBuffer();
		}
	}

	void GameConsole::present()
	{
		// Get and check console pointer
		auto consolePointer = CGL_internal::GConsole::get();
		if (consolePointer)
		{
			// Redirect call
			consolePointer->present();
		}
	}

	void GameConsole::putChar(unsigned int x, unsigned int y, char c)
	{
		// Get and check console pointer
		auto consolePointer = CGL_internal::GConsole::get();
		if (consolePointer)
		{
			// Redirect call
			consolePointer->putChar(x, y, c);
		}
	}

	void GameConsole::putString(unsigned int x, unsigned int y, const char* string, bool wrap)
	{
		// Get and check console pointer
		auto consolePointer = CGL_internal::GConsole::get();
		if (consolePointer)
		{
			// Redirect call
			consolePointer->putString(x, y, string, wrap);
		}
	}
}
